metrics.go

```diff
@@ -238,3 +240,19 @@ func protoMetricDescriptorTypeToMetricKind(m *metricspb.Metric) (googlemetricpb.
 		return googlemetricpb.MetricDescriptor_METRIC_KIND_UNSPECIFIED, googlemetricpb.MetricDescriptor_VALUE_TYPE_UNSPECIFIED
 	}
 }
+
+func protoResourceToMonitoredResource(rsp *resourcepb.Resource) *monitoredrespb.MonitoredResource {
+	if rsp == nil {
+		return nil
+	}
+	mrsp := &monitoredrespb.MonitoredResource{
+		Type: rsp.Type,
```

Comment:
nit: rsp.Type can also be nil.

Reply:
Type is a string though so can never be nil